<launch>
    <!-- Añadimos los nombres-->
    <arg name="odometry_topic" default="/odometry"/>
    <arg name="orientation_topic" default="/orientation"/>

    <!-- Añadir el resto de parametros para modificar -->

    <!-- Incluimos un archivo de lanzamiento llamado tfs.launch desde el paquete learning_ros_cpp -->
    <include file="$(find learning_ros_cpp)/launch/tfs.launch" />

    <!-- Creamos un nodo del paquete learning_ros_cpp con el ejecutable pkg_main y lo nombramos pkg_main. Mostramos la salida en la pantalla. -->
    <node pkg="learning_ros_cpp" type="pkg_main" name="pkg_main" output="screen">
        <param name="odometry_topic" value="$(arg odometry_topic)"/>
        <param name="orientation_topic" value="$(arg orientation_topic)"/>
    </node>
    
     <node pkg="learning_ros_cpp" type="publisher_node.py" name="publisher_node" output="screen">
        <param name="odometry_topic" value="$(arg odometry_topic)"/>
        <param name="orientation_topic" value="$(arg orientation_topic)"/>
    </node>
    <!-- Creamos un nodo del paquete rviz con el ejecutable rviz y lo nombramos rviz. Pasamos el argumento -d para especificar un archivo de configuración de rviz desde el paquete learning_ros_cpp -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find learning_ros_cpp)/config/Configuration.rviz" />

</launch>

